src/dash-table/components/ControlledTable/index.tsx

```diff
@@ -249,95 +280,68 @@ export default class ControlledTable extends PureComponent<ControlledTableProps>
-        let trs = r0c1.querySelectorAll('tr');
-        Array.from(r0c0.querySelectorAll('tr')).forEach((tr, index) => {
-            const tr2 = trs[index];
+        const currentTableWith = getComputedStyle(r1c1Table).width;
 
-            tr.style.height = `${tr2.clientHeight}px`;
-        });
+        if (!cycle) {
+            this.resizeFragmentTable(r0c0Table, currentTableWith);
+            this.resizeFragmentTable(r0c1Table, currentTableWith);
+            this.resizeFragmentTable(r1c0Table, currentTableWith);
+        }
 
-        // Adjust fixed columns headers to header's height
-        let trths = r1c1.querySelectorAll('tr > th:first-of-type');
-        Array.from(r1c0.querySelectorAll('tr > th:first-of-type')).forEach((th, index) => {
-            const tr2 = trths[index].parentElement as HTMLElement;
-            const tr = th.parentElement as HTMLElement;
+        if (fixed_columns || fixed_rows) {
+            const widths = Array.from(
+                r1c1.querySelectorAll('table.cell-table > tbody > tr:first-of-type > *')
+            ).map(c => c.getBoundingClientRect().width);
 
-            tr.style.height = getComputedStyle(tr2).height;
-        });
+            if (!cycle) {
+                this.resizeFragmentCells(r0c0, widths);
+                this.resizeFragmentCells(r0c1, widths);
+                this.resizeFragmentCells(r1c0, widths);
+            }
+        }
 
         if (fixed_columns) {
-            const r1c0Table = r1c0.querySelector('table') as HTMLElement;
-            const r1c1Table = r1c0.querySelector('table') as HTMLElement;
-
-            r1c0Table.style.width = getComputedStyle(r1c1Table).width;
-
-            const lastVisibleTd = r1c0.querySelector(`tr:first-of-type > *:nth-of-type(${fixed_columns})`);
-
-            let it = 0;
-            let currentWidth = r1c0.getBoundingClientRect().width;
-            let lastWidth = currentWidth;
-
-            do {
-                lastWidth = currentWidth
+            const lastFixedTd = r1c1.querySelector(`tr:first-of-type > *:nth-of-type(${fixed_columns})`);
+            if (lastFixedTd) {
+                const lastFixedTdBounds = lastFixedTd.getBoundingClientRect();
+                const lastFixedTdRight = lastFixedTdBounds.right - r1c1.getBoundingClientRect().left;
 
                 // Force first column containers width to match visible portion of table
-                if (lastVisibleTd) {
-                    const r1c0FragmentBounds = r1c0.getBoundingClientRect();
-                    const lastTdBounds = lastVisibleTd.getBoundingClientRect();
-                    currentWidth = lastTdBounds.right - r1c0FragmentBounds.left;
-
-                    const width = `${currentWidth}px`;
-
-                    r0c0.style.width = width;
-                    r1c0.style.width = width;
+                r0c0.style.width = `${lastFixedTdRight}px`;
+                r1c0.style.width = `${lastFixedTdRight}px`;
```

Comment:
Whether in a cycle or not, make sure the width allocated to the fixed columns is sufficient for the fixed columns.